### PR TITLE
[DOCS] Expand `updated` response parm in reindex API docs

### LINKED ISSUE #46438: Expand on `updated` parameter in Reindex API response body
The Reindex API documents the `update` parameter in the response body:

https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-reindex.html#docs-reindex-response-body

However, users have reached out wanting more detailed documentation on what is happening when documents show as "updated" in a reindex API operation.

We should expand the parameter definition.

CC @mark54g.

### PR DESCRIPTION
**Pull request by jrodewig**
The reindex API documentation's current `updated` response parameter definition doesn't explain what is happening when documents show as "updated" in a reindex.

This expands the parameter definition to capture the cases in which a document would be counted as `updated`.

Closes #46438.

### CODE REVIEW

File: docs/reference/docs/reindex.asciidoc [hunk 1]
```diff
@@ -522,9 +522,20 @@ reindex timed out.
 (integer) The number of documents that were successfully processed.
 
 `updated`::
-
++
+--
 (integer) The number of documents that were successfully updated.
```
Comment (henningandersen) [inline docs/reference/docs/reindex.asciidoc]: I would prefer to just clarify this line a bit and avoid the additional paragraphs.

```suggestion
(integer) The number of documents that were successfully updated, i.e., a document with same id already existed prior to reindex updating it.
```

**Review by henningandersen — COMMENTED** [review]
I added a comment inline. I think the additional failure scenario information should not go here, since it is quite advanced/technical. The problem is already documented here:

https://www.elastic.co/guide/en/elasticsearch/resiliency/current/index.html

under: "Better request retry mechanism when nodes are disconnected"

**Comment by jrodewig** [comment]
Thanks @henningandersen. I've updated the parm definition. Let me know if any other changes are needed.

**Review by henningandersen — APPROVED** [review]
LGTM, thanks @jrodewig

**Pull request merged**